ds3-sdk/src/main/java/com/spectralogic/ds3client/helpers/strategy/transferstrategy/AbstractTransferStrategy.java

```diff
@@ -167,30 +172,46 @@ private synchronized void maybeSetCachedException(final Throwable t) {
 
     private void transferJobParts(final Iterable<JobPart> jobParts,
                                   final CountDownLatch countDownLatch,
-                                  final AtomicInteger numBlobsTransferred) throws IOException {
+                                  final AtomicInteger numBlobsTransferred,
+                                  final Predicate transferPredicate) throws IOException
+    {
+        final AtomicInteger numJobPartsToTransfer = new AtomicInteger(Iterables.size(jobParts));
+
         for (final JobPart jobPart : jobParts) {
-            executorService.submit(new Callable<Void>() {
-                @Override
-                public Void call() throws Exception {
-                    try {
-                        transferMethod.transferJobPart(jobPart);
-                    } catch (final RuntimeException e) {
-                        emitFailureAndSetCachedException(e);
-                        throw e;
-                    } catch (final Exception e) {
-                        emitFailureAndSetCachedException(e);
-                        throw new RuntimeException(e);
-                    } finally {
-                        jobState.blobTransferredOrFailed(jobPart.getBlob());
-                        numBlobsTransferred.decrementAndGet();
-                        countDownLatch.countDown();
-                        return null;
-                    }
-                }
-            });
+            if (executorService.isShutdown()) {
+                countDownLatch.countDown();
+            } else {
+                maybeSubmitJobPartTransfer(countDownLatch, numBlobsTransferred, transferPredicate, numJobPartsToTransfer, jobPart);
+            }
         }
     }
 
+    private void maybeSubmitJobPartTransfer(final CountDownLatch countDownLatch, final AtomicInteger numBlobsTransferred, final Predicate transferPredicate, final AtomicInteger numJobPartsToTransfer, final JobPart jobPart) {
+        executorService.submit((Callable<Void>) () -> {
+            try {
+                if ( ! transferPredicate.test()) {
+                    countDownLatch.countDown();
+                } else {
+                    transferMethod.transferJobPart(jobPart);
+                }
+            } catch (final RuntimeException e) {
+                emitFailureAndSetCachedException(e);
+                throw e;
+            } catch (final Exception e) {
+                emitFailureAndSetCachedException(e);
+                throw new RuntimeException(e);
+            } finally {
```

Comment:
Seems like we're missing a throwable catch clause, unless we think that having both a runtime and checked exception handlers are enough.